www/assessment/items/itemsapi_distractors.php

```diff
@@ -57,79 +57,62 @@
-                itemsInstance.questions = getQuestionResponseIds;
-
-                $.each(itemsInstance.questions(), function(index, question) {
-                    question.on("validated", function() {
-                        var distractorQuestionLvl = this.getMetadata().distractor_rationale || "",
-                            distractorResponseLvl = this.getMetadata().distractor_rationale_response_level || "",
-                            validation = this.isValid("detailedWithPartials"),
-                            question = this.getQuestion(),
-                            outputHTML = "";
-
-                        if(!validation.correct) { // Could show 'correct' distractors
-                            if(distractorResponseLvl.length) {
-                                var response = this.getResponse(),
-                                    map = mapResponseLevel(question, response, validation),
-                                    list = "";
-
-                                $.each(distractorResponseLvl, function (index, distractor) {
-                                    if (map[index] === false) { // Change here to render unattempted
-                                        list += "<li>" + distractor + "</li>";
-                                    }
-                                });
-
-                                outputHTML += list.length ? "<ul>" + list + "</ul>" : "";
-                            } else if(distractorQuestionLvl.length) { // could render both level
-                                outputHTML += distractorQuestionLvl;
-                            }
-                            if(outputHTML == "" || outputHTML == null) {
-                                outputHTML = "Have you answered all possible responses?";
-                            }
-                            renderDistractor(question.response_id, outputHTML);
-                            MathJax.Hub.Queue(['Typeset', MathJax.Hub, question.response_id]);
-                        }
+        readyListener: function () {
+
+            itemsInstance.questionsApp().onEachQuestion('validated', function () {
+                var outputHTML = '';
+                var map, qid;
+                if (this.isValid()) {
+                    return;
+                }
+                if (this.getResponse().type === 'array') {
+                    map = this.mapValidationMetadata('distractor_rationale_response_level');
+                    $.each(map.incorrect, function (i, data) {
+
+                        // Each item in the `map.incorrect` array is an object
+                        // that contains a `value` key that holds the
+                        // response value, and a `metadata` key.
+                        var distractor = data.metadata;
+
+                        outputHTML += '<li>' + distractor + '</li>';
                     });
-                });
+                    if (outputHTML) {
```

Comment:
@jack-vo I think it's okay, it's nested within the `if` that starts on line 70. I normally wouldn't nest conditionals like this, but it's meant to be a basic demo